### ISSUE TITLE
"Failed to load the SVG string" log message

### ISSUE THREAD

**Issue by jayhlee** [nystudio107/craft-seomatic#452]
I keep getting a Warning Level log message stating "Failed to load the SVG string." The Category reads "craft\services\Assets::getAssetUrl"

I'm pretty sure the issue is related to SEOmatic because the log message disappears when I disable the plugin. I also stripped everything else on a test page so it's just a barebones Twig/HTML file, and the message still shows. Any idea what the culprit may be?

**Comment by khalwat** [comment]
I'd need more information to go on; maybe you have set something in SEOmatic to use an SVG asset that no longer exists?

**Comment by jayhlee** [comment]
I double checked, and I don't have any SVG asset that no longer exists. I also cleared all caches, but the warning log still shows up. Is there any other way to diagnose/pinpoint what asset SEOmatic is trying to unsuccessfully load?

**Comment by khalwat** [comment]
Give me more context in the log file -- the 10 lines before and after you see that message might help.

**Comment by jayhlee** [comment]
```
INNER JOIN `assets` `assets` ON `assets`.`id` = `elements`.`id`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `content` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)
WHERE (`elements_sites`.`siteId`='1') AND (`elements`.`id`=69) AND (`elements`.`archived`=FALSE) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)
ORDER BY `elements`.`dateCreated` DESC
LIMIT 1) `subquery`
INNER JOIN `assets` `assets` ON `assets`.`id` = `subquery`.`elementsId`
INNER JOIN `volumefolders` `volumeFolders` ON `assets`.`folderId` = `volumeFolders`.`id`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`
ORDER BY `elements`.`dateCreated` DESC
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][warning][craft\services\Assets::getAssetUrl] Failed to load the SVG string.
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][info][yii\db\Command::execute] DELETE FROM `assettransformindex` WHERE `id`=519
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile begin][yii\db\Command::execute] DELETE FROM `assettransformindex` WHERE `id`=519
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile end][yii\db\Command::execute] DELETE FROM `assettransformindex` WHERE `id`=519
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile end][craft\web\twig\Template::display] __string_template__0253d1f41a81df76c7d1bf827c5e66eb44b0a12b961518dffc2336e63dabf521
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile begin][craft\web\twig\Template::display] __string_template__dd33c8ca29fedce1f235935601f0109b7b85893071c38a988d065ad63317805b
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][info][yii\db\Command::query] SELECT `draftId`, `revisionId`
FROM `elements`
WHERE `id`=69
LIMIT 1
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile begin][yii\db\Command::query] SELECT `draftId`, `revisionId`
FROM `elements`
WHERE `id`=69
LIMIT 1
2019-08-09 11:11:15 [-][1][4n8tqhctdqppkj8s877l45dfp7][profile end][yii\db\Command::query] SELECT `draftId`, `revisionId`
FROM `elements`
WHERE `id`=69
LIMIT 1
```

**Comment by khalwat** [comment]
I'd need to see the 10 log messages (not lines) proceeding this error.

If you went to just send me the whole log file, that might be easier: andrew@nystudio107.com

**Comment by mattstein** [comment]
I've also just run into this problem and I don't think SEOmatic is at fault. 

@jayhlee probably had the issue disappear because a disabled SEOmatic no longer tried to transform the SVG into an appropriate social image format.

My problem seems to be with a specific SVG file—all others are transformed properly—and so far I've chased the issue down to [Craft/Imagine's `loadFromSVG()` failing to load the string](https://github.com/craftcms/cms/blob/develop/src/image/Raster.php#L498-L516). Same result using ImageMagick or GD.

Not sure of the cause yet, but my problematic SVG file...

- is only 85KB in size (not outrageous)
- has a proper DTD
- isn't any better after being opened and re-saved with Sketch
- isn't any better after being optimized+re-saved with [ImageOptim](https://imageoptim.com/mac)
- renders properly in the Chrome (directly and via `<img>` tag) and any Asset Previews

SEOmatic has the grace to fall back to the default social image, which is a nice touch. ❤

## tl;dr
 
Can't figure out what's so offensive about this one SVG, but I don't think SEOmatic is responsible.

**Comment by khalwat** [comment]
I'm pretty sure that's the case @mattstein thx

**Comment by mattstein** [comment]
Correction for science/posterity: looks like it's _any_ SVG, and caused by some regression either in Craft or Imagine.

**Comment by yoannisj** [comment]
I am bumping into a very similar (the same?) issue with front-end signup forms. Looks like SEOmatic is trying to transform the default `cpresources/user.svg` file associated to new users when submitting the signup form, which throws a server error:

```
2020-09-01 16:47:42 [-][-][hoertn2a78hsutierg8sg35q9n][error][craft\errors\ImageException] ImagickException: no decode delegate for this image format `SVG' @ error/blob.c/BlobToImage/353 in /home/runcloud/webapps/hotelcostes-web/vendor/pixelandtonic/imagine/src/Imagick/Imagine.php:135
Stack trace:
#0 /home/runcloud/webapps/hotelcostes-web/vendor/pixelandtonic/imagine/src/Imagick/Imagine.php(135): Imagick->readimageblob()
#1 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/image/Raster.php(509): Imagine\Imagick\Imagine->load()
#2 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/services/Images.php(197): craft\image\Raster->loadFromSVG()
#3 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/services/AssetTransforms.php(1528): craft\services\Images->loadImage()
#4 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/services/AssetTransforms.php(793): craft\services\AssetTransforms->_createTransformForAsset()
#5 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/services/AssetTransforms.php(684): craft\services\AssetTransforms->_generateTransform()
#6 /home/runcloud/webapps/hotelcostes-web/vendor/craftcms/cms/src/services/Assets.php(620): craft\services\AssetTransforms->ensureTransformUrlByIndexModel()
#7 /home/runcloud/webapps/hotelcostes-web/vendor/nystudio107/craft-seomatic/src/helpers/ImageTransform.php(130): craft\services\Assets->getAssetUrl()
#8 /home/runcloud/webapps/hotelcostes-web/vendor/nystudio107/craft-seomatic/src/services/Helper.php(422): nystudio107\seomatic\helpers\ImageTransform::socialTransform()
```

This may be due to a regression in Imagine / Craft with how it handles svg image transforms, but it only started happening when I installed SEOmatic, so maybe it's worth implementing a workaround / preflight check here?

**Comment by khalwat** [comment]
@yoannisj How is this SVG being passed into SEOmatic?

SEOmatic won't try to do a transform on an image unless you somehow pass it down to do so.

**Comment by yoannisj** [comment]
@khalwat Oh ok, I assumed seoMatic was listening to element save event to run the image transform, because this is happening when submitting a form to the `/users/save-user` action.

I see now that for this project user element's have preparse field which invokes `seomatic.meta.parsedValue('seoImage')` and gets rendered on save. Looking at the log it looks like the error is raised by social image transforms?

**Comment by khalwat** [comment]
@yoannisj somehow something is setting the `seoImage` to this SVG.

If you fix/change that, this error will go away

**Comment by yoannisj** [comment]
@khalwat indeed, the preparse field's twig was feeding the `seoImage` with an svg. I could solve the issue by ignoring svg images when setting the `seoImage` automatically in the field's twig code.

Anyway, sorry about the mixup and thanks for your help!
